From 06e0e47cab8cab09725dd850a65264f2f2317465 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Mon, 11 May 2015 18:09:35 -0700
Subject: [PATCH] HBASE-13412 ADDENDUM Region split decisions should have
 jitter

Hadoop 1 does not have Configuration#getDouble, use Configuration#getFloat
instead.
---
 .../hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java      | 4 ++--
 .../org/apache/hadoop/hbase/regionserver/TestRegionSplitPolicy.java   | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java
index 9ef7a9f..8bf4070 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java
@@ -52,8 +52,8 @@ public class ConstantSizeRegionSplitPolicy extends RegionSplitPolicy {
       this.desiredMaxFileSize = conf.getLong(HConstants.HREGION_MAX_FILESIZE,
         HConstants.DEFAULT_MAX_FILE_SIZE);
     }
-    double jitter = conf.getDouble("hbase.hregion.max.filesize.jitter", Double.NaN);
-    if (!Double.isNaN(jitter)) {
+    float jitter = conf.getFloat("hbase.hregion.max.filesize.jitter", Float.NaN);
+    if (!Float.isNaN(jitter)) {
       this.desiredMaxFileSize += (long)(desiredMaxFileSize * (RANDOM.nextFloat() - 0.5D) * jitter);
     }
   }
diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionSplitPolicy.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionSplitPolicy.java
index f24b0b8..73c9403 100644
--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionSplitPolicy.java
+++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionSplitPolicy.java
@@ -67,7 +67,7 @@ public class TestRegionSplitPolicy {
     // Configure IncreasingToUpperBoundRegionSplitPolicy as our split policy
     conf.set(HConstants.HBASE_REGION_SPLIT_POLICY_KEY,
       IncreasingToUpperBoundRegionSplitPolicy.class.getName());
-    conf.setDouble("hbase.hregion.max.filesize.jitter", 0.25D);
+    conf.setFloat("hbase.hregion.max.filesize.jitter", 0.25f);
 
     // Now make it so the mock region has a RegionServerService that will
     // return 'online regions'.
@@ -133,7 +133,7 @@ public class TestRegionSplitPolicy {
   @Test
   public void testCreateDefault() throws IOException {
     conf.setLong(HConstants.HREGION_MAX_FILESIZE, 1234L);
-    conf.setDouble("hbase.hregion.max.filesize.jitter", 0.25D);
+    conf.setFloat("hbase.hregion.max.filesize.jitter", 0.25f);
 
     // Using a default HTD, should pick up the file size from
     // configuration.
-- 
1.9.1

